File: android/src/main/java/com/kustom/mobile/sdk/reactnative/KustomMobileSDKPackage.java
package com.kustom.mobile.sdk.reactnative;

import androidx.annotation.NonNull;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import com.kustom.mobile.sdk.reactnative.checkout.KustomCheckoutViewManager;

import java.util.List;

public class KustomMobileSDKPackage implements ReactPackage {

    @NonNull
    @Override
    public List<NativeModule> createNativeModules(@NonNull ReactApplicationContext reactContext) {
        return List.of();
    }

    @NonNull
    @Override
    public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactContext) {
        return List.of(
                new KustomCheckoutViewManager(reactContext)
        );
    }
}

File: android/src/main/java/com/kustom/mobile/sdk/reactnative/KustomMobileSDKPackage.kt
package com.kustom.mobile.sdk.reactnative

import com.facebook.react.BaseReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfoProvider
import com.facebook.react.uimanager.ViewManager
import com.kustom.mobile.sdk.reactnative.checkout.KustomCheckoutViewManager

class KustomMobileSDKPackage : BaseReactPackage() {
  override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
    return listOf(KustomCheckoutViewManager())
  }

  override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? = null

  override fun getReactModuleInfoProvider() = ReactModuleInfoProvider { emptyMap() }
}
